contracts/ROSCA.sol

```diff
@@ -162,8 +219,14 @@ contract ROSCA {
    * Any excess funds will be withdrawable through withdraw().
    */
   function contribute() payable {
-    if (!members[msg.sender].alive) throw;
+    if (!members[msg.sender].alive || currentRound == 0) throw;
     members[msg.sender].contributed += msg.value;
+    members[msg.sender].goodStanding = members[msg.sender].contributed / contributionSize;
+    if(members[msg.sender].contributed > contributionSize * membersAddresses.length) {
+      uint excessContribution = members[msg.sender].contributed - (contributionSize * membersAddresses.length);
+      members[msg.sender].contributed -= excessContribution;
```

Comment:
This is correct logic, but I think our code is getting too complex with "contributed", "goodStanding" and "pendingWithdrawl".

What if we do the following: we rename "pendingWithdrawl" to "credit" and make it a signed int.
* When someone sends a contribution, we add it to "credit"
* At end of round, we deduct contributionSize from the "credit" as many participants as possible, and add lowestBid to the winner's "credit"
* When someone wants to bid, we first validate that they have nonnegative "credit"
* When someone wants to withdraw, they can withdraw up to "credit" (or zero if negative)

What do you think?